kore/test/Test/Debug.hs

```diff
@@ -208,3 +228,25 @@ test_Debug =
     ]
   where
     yields input = Terse.equals (show $ debug input)
+
+test_diff :: [TestTree]
+test_diff =
+    [ test (SA A, SA A)            $ Nothing
+    , test (SA A, SB B)            $ Just "{- was: SA A -} SB B"
+    , test (N (SA A), N (SB B))    $ Just "N {- was: (SA A) -} (SB B)"
+    , test (Rn { unRn = SA A }, Rn { unRn = SB B })
+        $ Just "Rn { unRn = {- was: SA A -} SB B }"
+    , test ("A" :: String, "B")    $ Just "{- was: \"A\" -} \"B\""
+    , test ('A', 'B')              $ Just "{- was: 'A' -} 'B'"
+    , test (0 :: Integer, 1)       $ Just "{- was: 0 -} 1"
+    , test (True, False)           $ Just "{- was: True -} False"
+    , test ([True], [])            $ Just "{- was: [ True ] -} []"
+    , test ([], [True])            $ Just "{- was: [] -} [ True ]"
+    , test ([True], [True, False]) $ Just "_ : {- was: [] -} [ False ]"
```

Comment:
Could you add the following tests? Or, at least, the ones that make sense.
* `test ([True, True], [True, False, True])`
* `test (Struct {field1=True, field2=True, field3=True}, Struct {field1=False, field2=False, field3=True})`
* `test (Struct {field1=True, field2=True, field3=True}, Struct {field1=True, field2=False, field3=False})`
* A test with a constructor operator like `:<`